spec/features/editor_spec.rb

```diff
@@ -93,6 +94,22 @@
     end
   end
 
+  context 'when an exercise has one or more teacher_defined_tests' do
+    it 'displays the score button' do
+      visit(implement_exercise_path(exercise))
+      expect(page).to have_content(exercise.title)
+      expect(page).to have_css('#assess')
+    end
+  end
+
+  context 'when an exercise has no teacher_defined_tests' do
+    it 'disables the score button' do
+      visit(implement_exercise_path(exercise_without_test))
+      expect(page).to have_content(exercise_without_test.title)
+      expect(page).not_to have_css('#assess')
+    end
+  end
+
```

Comment:
This is the cleanest solution I found, if you have any advices on making it cleaner, I would love to hear your thoughts.